### DISCUSSION TITLE
Docker, net.tcp and CoreWCF

### DISCUSSION THREAD

**Discussion by oskarjf** [CoreWCF/CoreWCF#1255] [Q&A]
Hi

I'm wondering if anyone has had this issue that I'm experiencing.
If I run my solution in Visual Studio my HTTP and NET.TCP endpoints work fine.
When running my solution within a Docker container locally I get the error 

> System.ServiceModel.EndpointNotFoundException: The message with To 'net.tcp://localhost:8087/Integration/Services/UserService.svc' cannot be processed at the receiver, due to an AddressFilter mismatch at the EndpointDispatcher.  Check that the sender and receiver's EndpointAddresses agree.

Does it have any effect that I'm using the System.ServiceModel.ServiceContractAttribute rather than CoreWCF.ServiceContractAttribute?

**Comment by oskarjf** [comment]
**My endpoints are defined like this:**

```
var netTcpBinding = new NetTcpBinding()
{
    MaxReceivedMessageSize = 33554432,
    Security = new NetTcpSecurity() { Mode = SecurityMode.None },
    ReaderQuotas = new System.Xml.XmlDictionaryReaderQuotas() { MaxDepth = 100, MaxBytesPerRead = 209715200, MaxStringContentLength = 209715200, MaxArrayLength = 209715200 },
    Name = "netTCPBinding"
};

builder.AddService<CapabilityQueryService>((serviceOptions) =>
{
    serviceOptions.DebugBehavior.IncludeExceptionDetailInFaults = includeExceptionDetailInFaults;
    serviceOptions.BaseAddresses.Add(new Uri($"http://{myServerAndPort}/Integration/Services/CapabilityQueryService.svc"));
})
.AddServiceEndpoint<CapabilityQueryService, IQueryCapabilityDomain>(httpBinding, string.Empty)
.AddServiceEndpoint<CapabilityQueryService, IQueryCapabilityDomain>(netTcpBinding, $"net.tcp://{tcpServerAndPort}/Integration/Services/CapabilityQueryService.svc");
```

**Reply by gyurisc** [reply]
I am seeing a similar issue. It works locally but not in docker. Did you manage to figure out this one?
